.editorconfig

```
@@ -0,0 +1,13 @@
+# Copyright (C) Daniel Stenberg, <daniel@haxx.se>, et al.
+#
+# SPDX-License-Identifier: curl
+
+root = true
+
+[*.{c,h}]
+charset = utf-8
+insert_final_newline = true
+indent_style = space
+indent_size = 2
+max_line_length = 79
+trim_trailing_whitespace = true
```

```suggestion
[*]
charset = utf-8
insert_final_newline = true
indent_style = space
trim_trailing_whitespace = true

[*.{c,h}]
indent_size = 2
max_line_length = 79

[*.{pl,pm}]
indent_size = 4
```
maybe? (Though not all files have an extension.)

---

I usually avoid doing global rules cause some stuff like bash files can be picky when it comes to formatting, so I prefer to only set it for known extensions.

---

These global rules are already enforced via the spacecheck script,
it seems safe to to, if opt for having an editorconfig in addition to
that.